backend/api/v1/auth_service.go

```diff
@@ -284,6 +295,53 @@ func (s *AuthService) UpdateUser(ctx context.Context, request *v1pb.UpdateUserRe
 		passwordHashStr := string(passwordHash)
 		patch.PasswordHash = &passwordHashStr
 	}
+	if request.MfaCode != nil {
+		secret := user.MFAConfig.TempOtpSecret
+		isValid := otp.ValidateWithCodeAndSecret(*request.MfaCode, secret)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to validate MFA code, error: %v", err)
+		}
+		if !isValid {
+			return nil, status.Errorf(codes.InvalidArgument, "invalid MFA code")
+		}
+		patch.MFAConfig = &storepb.MFAConfig{
+			OtpSecret:         user.MFAConfig.TempOtpSecret,
+			TempOtpSecret:     user.MFAConfig.TempOtpSecret,
```

Comment:
Any reason to set TempOtpSecret and TempRecoveryCodes again?

Reply:
Removed